### PR TITLE
Fix for use case when playback starts with the seek.

### PR DESCRIPTION
**Pull request by tomkordic**
Previous behavior seeking, seeked, playing, this caused some errors in calculations on backend.
Current behaviour play, seeking, seeked, playing.

I hope this is correct.

Use case:

Playback starts from 1/3 of the content, play when ready true.

### CODE REVIEW

File: MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java [hunk 1]
```diff
@@ -237,6 +244,9 @@ public void enableMuxCoreDebug(boolean enable, boolean verbose) {
 
   @SuppressWarnings("unused")
   public void videoChange(CustomerVideoData customerVideoData) {
+    numberOfPauseEventsSent = 0;
```
Comment (jsanford8) [inline MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java]: You'll want to reset these in `programChange`, too

File: MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java [hunk 2]
```diff
@@ -448,9 +465,14 @@ protected void pause() {
...
   protected void play() {
-    if (state == PlayerState.REBUFFERING
+    // If this is the first play event it may be very important not to be skipped
+    // In all other cases skip this play event
+    if (
+        (state == PlayerState.REBUFFERING
         || seekingInProgress
-        || state == PlayerState.SEEKED) {
+        || state == PlayerState.SEEKED) &&
+            (numberOfPlayEventsSent > 0)
```
Comment (jsanford8) [inline MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java:473]: I like this part

File: MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java [hunk 3]
```diff
@@ -432,7 +449,7 @@ protected void buffering() {
   }
 
   protected void pause() {
-    if (state == PlayerState.SEEKED) {
+    if (state == PlayerState.SEEKED && numberOfPauseEventsSent > 0) {
```
Comment (jsanford8) [inline MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java:451]: This feels a bit risky. Is this necessary? What does it help address?

**Review by jsanford8 — CHANGES_REQUESTED** [review]
One thing to change, and then a question about another part.

File: MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java (hunk 1 above)
Comment (tomkordic) [reply MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java]: OK

File: MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java (hunk 3 above)
Comment (tomkordic) [reply MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java:451]: Same as with play event, in general we ignore this event if seeking is in progress, there is a specific case when seek is set but play when ready is false we should send a pause event right after the seeked event, in every other case we should ignore pause event.

File: MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java (hunk 1 above)
Comment (tomkordic) [reply MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java]: Upldated

File: MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java [hunk 4]
```diff
@@ -285,6 +294,13 @@ public void setStreamType(int type) {
   @Override
   public void dispatch(IEvent event) {
     if (player != null && player.get() != null && muxStats != null) {
+      numberOfEventsSent ++;
```
Comment (skidder) [inline MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java]: Can you remove the whitespace between the variable name and the `++`? This should be handled by the code formatter.

**Review by skidder — APPROVED** [review]
Approved, but please run the code formatter to address some style issues.

**Review by jsanford8 — APPROVED** [review]

**Pull request merged**